src/utils/astFromValue.js

```diff
@@ -0,0 +1,129 @@
+/* @flow */
+/**
+ *  Copyright (c) 2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+import invariant from '../utils/invariant';
+import isNullish from '../utils/isNullish';
+import type { Value } from '../language/ast';
+import {
+  INT,
+  FLOAT,
+  STRING,
+  BOOLEAN,
+  ENUM,
+  LIST,
+  OBJECT,
+  OBJECT_FIELD,
+} from '../language/kinds';
+import {
+  GraphQLType,
+  GraphQLEnumType,
+  GraphQLInputObjectType,
+  GraphQLList,
+  GraphQLNonNull,
+} from '../type/definition';
+import { GraphQLFloat } from '../type/scalars';
+
+
+/**
+ * Produces a GraphQL Value AST given a JavaScript value.
+ *
+ * Optionally, a GraphQL type may be provided, which will be used to
+ * disambiguate between value primitives.
+ *
+ * | JSON Value    | GraphQL Value        |
+ * | ------------- | -------------------- |
+ * | Object        | Input Object         |
+ * | Array         | List                 |
+ * | Boolean       | Boolean              |
+ * | String        | String / Enum Value  |
+ * | Number        | Int / Float          |
+ *
+ */
+export default function astFromValue(
```

Comment:
I think this function is sufficiently complicated that it should have its own unit tests. Would be useful as documentation as well.